src/CMakeLists.txt

```diff
@@ -628,31 +628,40 @@ install(TARGETS ceph-mon DESTINATION bin)
 
 # OSD/ObjectStore
 # make rocksdb statically
-if(NOT ALLOCATOR STREQUAL "jemalloc")
-  set(disable_jemalloc "DISABLE_JEMALLOC=1")
-endif()
 
-set(ROCKSDB_EXTRA_CXXFLAG -fPIC -Wno-unused-variable)
-if(NOT CMAKE_BUILD_TYPE STREQUAL Debug)
-  set(ROCKSDB_EXTRA_CXXFLAG ${ROCKSDB_EXTRA_CXXFLAG} -DNDEBUG)
-endif(NOT CMAKE_BUILD_TYPE STREQUAL Debug)
+set(ROCKSDB_CXX_FLAGS "-DCMAKE_CXX_FLAGS=-fPIC")
+
+if(ALLOCATOR STREQUAL "jemalloc")
+  set(ROCKSDB_JEMALLOC "-DWITH_JEMALLOC=ON")
+endif()
 
-set(ROCKSDB_CXX "${CMAKE_CXX_COMPILER}")
 if (WITH_CCACHE AND CCACHE_FOUND)
-  set(ROCKSDB_CXX "ccache" "${CMAKE_CXX_COMPILER}")
+	  set(ROCKSDB_CXX "-DCMAKE_CXX_COMPILER=ccache")
```

Comment:
we are using indent width of 2 characters in CMake scripts, following the coding style of cmake official modules.